### PR TITLE
fix(react-infobutton): Add aria-owns to InfoLabel

### PR DESCRIPTION
**Pull request by sopranopillow**
## Previous Behavior

`aria-owns` is not set, therefore the popup is not linked to the trigger.

## New Behavior

`aria-owns` is now set by InfoLabel and a Do has been added advising to add `aria-owns` to the parent of the trigger.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                      |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                 Change |
| :----------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoButton.fixture.js'>InfoButton</abbr> | `127.977 kB`<br />`39.012 kB` | `128.022 kB`<br />`39.047 kB` | `45 B` <br />`35 B`  |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoLabel.fixture.js'>InfoLabel</abbr>   |      `131.26 kB`<br />`40 kB` | `131.306 kB`<br />`40.023 kB` | `46 B` <br />`23 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `64.855 kB`<br />`17.852 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `203.937 kB`<br />`57.086 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |    `36.086 kB`<br />`11.9 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.446 kB`<br />`2.186 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/79c08ce5bbf6387b2b18ba4c3d2d5681e0177d4b'>79c08ce5bbf6387b2b18ba4c3d2d5681e0177d4b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/InfoButton-mount_base.html">24</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/InfoButton-mount.html">24</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/Avatar-mount_base.html">800</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/Avatar-mount.html">835</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/Button-mount_base.html">419</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/Button-mount.html">421</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/Field-mount_base.html">1455</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/Field-mount.html">1421</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/FluentProvider-mount_base.html">1026</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/FluentProvider-mount.html">997</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/FluentProviderWithTheme-mount_base.html">128</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/FluentProviderWithTheme-mount.html">145</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">107</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">112</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">128</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">118</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/InfoButton-mount_base.html">24</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/InfoButton-mount.html">24</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/MakeStyles-mount_base.html">1317</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/MakeStyles-mount.html">1377</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/Persona-mount_base.html">2319</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/Persona-mount.html">2322</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/SpinButton-mount_base.html">1767</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27834/perf-test-react-components/SpinButton-mount.html">1765</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts [hunk 1]
```diff
@@ -55,6 +56,7 @@ export const useInfoLabel_unstable = (props: InfoLabelProps, ref: React.Ref<HTML
 
   if (infoButton) {
     infoButton['aria-labelledby'] ??= `${label.id} ${infoButton.id}`;
+    root['aria-owns'] = (infoButton.info as InfoButtonProps).id;
```
Comment (behowell) [inline packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts:59]: You'll need to resolve the shorthand for the info slot, and generate this ID:

```ts
const infoButtonInfo = resolveShorthand(infoButton.info, {
  defaultProps: {
   id: generatedInfoId, // from a useId call outside this if statement
  }
});
infoButton.info = infoButtonInfo;
root['aria-owns'] = infoButtonInfo.id;
```

File: packages/react-components/react-infobutton/stories/Infobutton/InfoButtonBestPractices.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ Best Practices
 
 - Prefer using an `InfoLabel` if the `InfoButton` is intended to be associated with a label.
 - Set `aria-label` to an appropriate value if the `InfoButton` is not associated with a label.
+- Add `aria-owns` to the trigger's parent element using the id of the `info` slot.
```
Comment (behowell) [inline packages/react-components/react-infobutton/stories/Infobutton/InfoButtonBestPractices.md:10]: Include something like "This is done automatically if using `InfoLabel`."

File: packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts (hunk 1 above)
Comment (behowell) [reply packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts:59]: Also, you don't want to overwrite any existing aria-owns prop, Wrap that all with `if (root['aria-owns'] === undefined) {`

File: packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts [hunk 3]
```diff
@@ -48,13 +48,24 @@ export const useInfoLabel_unstable = (props: InfoLabelProps, ref: React.Ref<HTML
...
+  });
+
+  const infoButtonInfo = resolveShorthand(infoButton?.info, {
+    defaultProps: {
+      id: useId('infobutton-info-'),
     },
   });
 
   if (infoButton) {
+    infoButton.info = infoButtonInfo;
```
Comment (behowell) [inline packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts:63]: nit: If you take my `baseId` suggestion, then this call to resolveShorthand can move inside the `if (infoButton)`:
```suggestion
  if (infoButton) {
    infoButton.info = resolveShorthand(infoButton.info, {
      defaultProps: {
        id: baseId + '__info',
      },
    });
```

File: packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts [hunk 4]
```diff
@@ -48,13 +48,24 @@ export const useInfoLabel_unstable = (props: InfoLabelProps, ref: React.Ref<HTML
...
       id: useId('infobutton-'),
-      info,
       size,
+      info,
+    },
+  });
+
+  const infoButtonInfo = resolveShorthand(infoButton?.info, {
+    defaultProps: {
+      id: useId('infobutton-info-'),
```
Comment (behowell) [inline packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts:58]: This is optional to do, but in the case of a single element having several generated IDs, I prefer to generate a base ID and then append the slot name to it. This both helps debug (easier to see where the ID came from in dev tools), and reduces the number of calls to `useId`. (See `useField` for an example)

Like:
```ts
const baseId = useId('infolabel-');

// For the label slot:
id: baseId + '__label',

// For the infoButton slot:
id: baseId + '__infoButton',

// For the info slot:
id: baseId + '__info',
```

File: packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts [hunk 5]
```diff
@@ -48,13 +48,24 @@ export const useInfoLabel_unstable = (props: InfoLabelProps, ref: React.Ref<HTML
...
     },
   });
 
   if (infoButton) {
+    infoButton.info = infoButtonInfo;
     infoButton['aria-labelledby'] ??= `${label.id} ${infoButton.id}`;
+
+    if (infoButton['aria-owns'] === undefined) {
+      root['aria-owns'] = infoButton.info?.id;
+    }
```
Comment (behowell) [inline packages/react-components/react-infobutton/src/components/InfoLabel/useInfoLabel.ts:68]: This needs to check `root`, not `infoButton`:
```suggestion
    if (root['aria-owns'] === undefined) {
      root['aria-owns'] = infoButton.info?.id;
    }
```

Or you could skip the if statement and use `??=` (like with the `aria-labelledby` above):
```suggestion
    root['aria-owns'] ??= infoButton.info?.id;
```

**Review by behowell — APPROVED** [review]
One correctness fix (need to check `root['aria-owns']` instead), and two optional suggestions. Otherwise looks good, thanks!

**Pull request merged**